src/RingGrids/general.jl

```diff
@@ -0,0 +1,324 @@
+Provide grid `Grid` and its resolution parameter `nlat_half`. For both_hemisphere==false only
+the northern hemisphere (incl Equator) is returned."""
+function get_nlons(Grid::Type{<:AbstractGridArray}, nlat_half::Integer; both_hemispheres::Bool=false)
+    n = both_hemispheres ? get_nlat(Grid, nlat_half) : nlat_half
+    return [get_nlon_per_ring(Grid, nlat_half, j) for j in 1:n]
+end
+
+## ITERATORS
+"""
+$(TYPEDSIGNATURES)
+CartesianIndices for the 2nd to last dimension of an AbstractGridArray,
+to be used like
+
+for k in eachgrid(grid)
+    for ring in eachring(grid)
+        for ij in ring
+            grid[ij, k]"""
+@inline eachgrid(grid::AbstractGridArray) = CartesianIndices(size(grid)[2:end])
+
+"""
+$(TYPEDSIGNATURES)
+Vector{UnitRange} `rings` to loop over every ring of grid `grid`
+and then each grid point per ring. To be used like
+
+    rings = eachring(grid)
+    for ring in rings
+        for ij in ring
+            grid[ij]"""
+@inline eachring(grid::AbstractGridArray) = grid.rings
+
+function eachring(Grid::Type{<:AbstractGridArray}, nlat_half::Integer)
+    rings = Vector{UnitRange{Int}}(undef, get_nlat(Grid, nlat_half))
+    each_index_in_ring!(rings, Grid, nlat_half)
+    return rings
+end
+
+"""
+$(TYPEDSIGNATURES)
+Same as `eachring(grid)` but performs a bounds check to assess that all grids
+in `grids` are of same size."""
+function eachring(grid1::Grid, grids::Grid...) where {Grid<:AbstractGridArray}
+    @inline 
+    n = length(grid1)
+    Base._all_match_first(X->length(X), n, grid1, grids...) || throw(BoundsError)
+    return eachring(grid1)
+end
+
+function grids_match(A::AbstractGridArray, B::AbstractGridArray)
+    length(A) == length(B) && return grids_match(typeof(A), typeof(B))
+    return false
+end
+
+function grids_match(A::Type{<:AbstractGridArray}, B::Type{<:AbstractGridArray})
+    return nonparametric_type(A) == nonparametric_type(B)
+end
+
+"""
+$(TYPEDSIGNATURES)
+UnitRange to access data on grid `grid` on ring `j`."""
+function each_index_in_ring(grid::Grid, j::Integer) where {Grid<:AbstractGridArray}
+    return each_index_in_ring(Grid, j, grid.nlat_half)
+end
+
+"""
+$(TYPEDSIGNATURES)
+UnitRange to access each grid point on grid `grid`."""
+eachgridpoint(grid::AbstractGridArray) = Base.OneTo(get_npoints(grid))
+function eachgridpoint(grid1::Grid, grids::Grid...) where {Grid<:AbstractGridArray}
+    n = length(grid1)
+    Base._all_match_first(X->length(X), n, grid1, grids...) || throw(BoundsError)
+    return eachgridpoint(grid1)
+end
+
+"""
+$(TYPEDSIGNATURES)
+Obtain ring index j from gridpoint ij and Vector{UnitRange} describing rind indices
+as obtained from eachring(::Grid)"""
+function whichring(ij::Integer, rings::Vector{UnitRange{Int}})
+    @boundscheck 0 < ij <= rings[end][end] || throw(BoundsError)
+    j = 1
+    @inbounds while ij > rings[j][end]
+        j += 1
+    end
+    return j
+end
+
+## BROADCASTING
```

Comment:
Is there currently a problem with the broadcasting? I think in general, it should be much simpler than for `LowerTriangularArray`, as we really only need to specialise the `similar` function.

Reply:
Nah, when you reviewed this I just hadn't worked on the broadcasting yet, seems to work fine. I had the feeling that the `if isstructurepreserving(bc) || fzeropreserving(bc)` wasn't needed for RingGrids, so I skipped that. If you think it is, please elaborate! 💚